src/abstractprobprog.jl

```diff
@@ -60,3 +61,23 @@ m = decondition(condition(m, obs))
 should hold for generative models `m` and arbitrary `obs`.
 """
 function condition end
+
+
+"""
+    rand([rng=Random.default_rng()], [T=NamedTuple], model::AbstractProbabilisticProgram) -> T
+
+Draw a sample from the prior and prior-predictive distribution of the model specified by the
+probabilistic program.
```

Comment:
Prior-predictive distribution is not completely correct, is it? AFAIK the prior-predictive distribution is the distribution of observation averaged over the parameters with respect to the prior, but in `rand` we do not perform this integration but rather sample observations for a specific set of sampled parameters. Maybe the more correct term would be
```suggestion
Draw a sample from the joint distribution of the model specified by the probabilistic program.
```